examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,287 @@
+#
+# .. math::
+#   |+_{n}\rangle = \frac{1}{\sqrt{2^n}}\sum_{z=0}^{2^n-1} |z\rangle
+#
+#
+# We aim to explore the space of bitstring states for a superposition which is likely to yield a
+# large value for the :math:`C` operator upon performing a measurement in the computational basis.
+# Using the :math:`2p` angle parameters
+# :math:`\boldsymbol{\gamma} = \gamma_1\gamma_2...\gamma_p`, :math:`\boldsymbol{\beta} = \beta_1\beta_2...\beta_p`
+# we perform a sequence of operations on our initial state:
+#
+# .. math::
+#   |\boldsymbol{\gamma},\boldsymbol{\beta}\rangle = U_{B_p}U_{C_p}U_{B_{p-1}}U_{C_{p-1}}...U_{B_1}U_{C_1}|+_n\rangle
+#
+# where the operators have the explicit forms
+#
+# .. math::
+#   U_{B_i} &= e^{-i\beta_iB} = \prod_{j=1}^n e^{-i\beta_i\sigma_x^j} \\
+#   U_{C_i} &= e^{-i\gamma_iC} = \prod_{\text{edge (j,k)}} e^{-i\gamma_i(1-\sigma_z^j\sigma_z^k)/2}
+#
+# In other words, we make :math:`p` layers of our parametrized :math:`U_bU_C` gates.
+# These can be implemented on a quantum circuit using the gates depicted below (up to an irrelevant constant
+# that gets absorbed into the parameters).
+#
+# .. figure:: ../../examples/figures/qaoa_operators.png
+#    :align: center
+#    :scale: 100%
+#    :alt: qaoa_operators
+#
+# |
+# Let :math:`\langle \boldsymbol{\gamma},
+# \boldsymbol{\beta} | C | \boldsymbol{\gamma},\boldsymbol{\beta} \rangle` be the expectation of our objective.
+# In the next section, we will use PennyLane to sample this expectation value and perform classical optimization
+# over the parameters. This will specify a state :math:`|\boldsymbol{\gamma},\boldsymbol{\beta}\rangle` which is
+# likely to yield an approximately optimal partition :math:`|z\rangle` upon performing a measurement in the
+# computational basis.
+# In the case of the graph shown above, we want to measure either 0101 or 1010 from our state.
+#
+# .. figure:: ../../examples/figures/qaoa_optimal_state.png
+#   :align: center
+#   :scale: 60%
+#   :alt: optimal_state
+#
+#
+# Implementing QAOA in PennyLane
+# ------------------------------
+#
+# Imports and setup
+# ~~~~~~~~~~~~~~~~~
+#
+# To get started, we import PennyLane along with the PennyLane-provided
+# version of NumPy.
+
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+##############################################################################
+# Operators
+# ~~~~~~~~~
+# We specify the number of qubits (vertices) with ``n_wires`` and
+# compose the unitary operators using the definitions
+# above. :math:`U_B` operators act on individual wires, while :math:`U_C`
+# operators act on wires whose corresponding vertices are joined by an edge in
+# the graph. We also define the graph using
+# the list ``graph``, which contains the tuples of vertices defining
+# each edge in the graph.
+
+n_wires = 4
+graph = [(0, 1), (0, 3), (1, 2), (2, 3)]
+
+# unitary operator U_B with parameter beta
+def U_B(beta):
+    for wire in range(n_wires):
+        qml.RX(2 * beta, wires=wire)
```

Comment:
This factor of 2 accompanying the beta parameter doesn't seem to match the description of `U_B` from above. I guess it's because of the way `qml.RX()` is defined, but worth explaining briefly here.

Reply:
Ah good catch. Actually all of the factors can be absorbed into the parameters but in this case it would be helpful to include the factor of 2 in the figure above, since this would exactly describe `U_B`. The way `qml.RX()` is defined is consistent with the RX in the diagram.